packages/sdks/snippets/react-sdk-next-pages/src/pages/product-details/product/category/[handle].tsx

```diff
@@ -0,0 +1,43 @@
+import { fetchOneEntry, type BuilderContent } from '@builder.io/sdk-react';
+import Image from 'next/image';
+
+export default function ProductCategoryPage({
+  productDetails,
+}: {
+  productDetails: BuilderContent;
+}) {
+  return (
+    <div className="product-details">
+      <h1>{productDetails.data?.name}</h1>
+      <Image
+        src={productDetails.data?.image}
+        alt={productDetails.data?.name}
+        width="400"
+        height="500"
+        priority={true}
+      />
+      <p>{productDetails.data?.collection.value.data.copy}</p>
+      <p>Price: {productDetails.data?.collection.value.data.price}</p>
+    </div>
+  );
+}
+
+export const getServerSideProps = async (context: {
+  params: { handle: string };
+}) => {
```

Comment:
uses `GetServerSideProps` type, and then use `InferGetServerSidePropsType` to pass the prop to the component


https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props#example